hf_xet/src/runtime.rs

```diff
@@ -30,17 +30,57 @@ fn install_sigint_handler() -> Result<(), MultithreadedRuntimeError> {
     Ok(())
 }
 
+#[cfg(windows)]
+extern "system" fn console_ctrl_handler(
+    ctrl_type: winapi::shared::minwindef::DWORD,
+) -> winapi::shared::minwindef::BOOL {
+    use winapi::um::wincon;
+
+    // Only handle CTRL_C_EVENT
+    if ctrl_type == wincon::CTRL_C_EVENT {
+        // Check if we have active operations
+        let has_active_ops = {
+            let guard = MULTITHREADED_RUNTIME.read().unwrap();
+            if let Some((runtime_pid, ref runtime)) = *guard {
+                runtime_pid == std::process::id() && runtime.external_executor_count() > 0
+            } else {
+                false
+            }
+        };
+
+        if has_active_ops {
+            // We have active operations, handle it ourselves
+            SIGINT_DETECTED.store(true, Ordering::SeqCst);
+            winapi::shared::minwindef::TRUE
```

Comment:
This is the key thing here.  We always want to return FALSE.  Returning true blocks the python signal handler from running as well.